token/program-2022/src/extension/confidential_transfer/mod.rs

```diff
@@ -0,0 +1,34 @@
+use {
+    crate::extension::{AccountType, Extension, ExtensionType},
+    bytemuck::{Pod, Zeroable},
+};
+
+/// Confidential Transfer Extension instructions
+pub mod instruction;
+
+/// Confidential Transfer Extension processor
+pub mod processor;
+
+/// Transfer auditor configuration
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+pub struct MintConfidentialTransferAuditor {
+    // TODO: inline `zk_token_program::state::Auditor` here
+}
+
+impl Extension for MintConfidentialTransferAuditor {
+    const TYPE: ExtensionType = ExtensionType::MintConfidentialTransferAuditor;
+    const ACCOUNT_TYPE: AccountType = AccountType::Mint;
+}
+
+/// Confidential account state
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+pub struct AccountConfidentialState {
```

Comment:
nit: how about `AccountConfidentialTransferState` or `AccountConfidentialTransfer` for consistency?

Reply:
I struggled with this and found the convention of a `Mint` or `Account` prefix made this naming tough.

I had "Transfer" in there for a while actually.  I settled on removing it because the tokens are being held confidentially in this structure, and adding "Transfer" to me implied that a transfer operation was pending.

I'm not sure I like of the options, even the existing name in the PR, too much frankly

Reply:
`AccountConfidentialTransferAccountState` feels the most accurate, but that 2x Account is no good.

I like naming extensions `ConfidentialTransferAccountState` and `ConfidentialTransferAuditor`, which breaks the `Mint`/`Account` convention. I think I'd get over that though!

Reply:
> I like naming extensions ConfidentialTransferAccountState and ConfidentialTransferAuditor

I pushed a new commit, 6eaa6d6, with these to show how that'd look

Reply:
Why exactly do we need that `Account` in there between `Transfer` and `State`?

Reply:
I actually don't mind losing the Mint/Account prefixes, since `ACCOUNT_TYPE` is defined for every Extension.
This is a radical suggestion that I'm not sure I love, but we could actually strip those prefixes from the existing extensions as well, leaving something like:
```
pub struct TransferFeeConfig {
    ...
}
impl Extension for TransferFeeConfig {
    const TYPE: ExtensionType = ExtensionType::TransferFeeConfig;
    const ACCOUNT_TYPE: AccountType = AccountType::Mint;
}

pub struct TransferFeeAmount {
    /// Amount withheld during transfers, to be harvested to the mint
    pub withheld_amount: PodU64,
}
impl Extension for TransferFeeAmount {
    const TYPE: ExtensionType = ExtensionType::TransferFeeAmount;
    const ACCOUNT_TYPE: AccountType = AccountType::Account;
}
```

```
pub struct CloseAuthority {
    /// Optional authority to close a mint
    pub close_authority: OptionalNonZeroPubkey,
}
impl Extension for CloseAuthority {
    const TYPE: ExtensionType = ExtensionType::CloseAuthority;
    const ACCOUNT_TYPE: AccountType = AccountType::Mint;
}
```

Reply:
> Why exactly do we need that `Account` in there between `Transfer` and `State`?

Not sure! `ConfidentialTransferState` definitely feels better.

I like your radical suggestion, with the exception of `CloseAuthority` as this is now ambiguous due to how an Account already has a close authority.  In this case `MintCloseAuthority` seems quite appropriate.

Reply:
> I like your radical suggestion

I'll PR it so we can take a look

Reply:
> I actually don't mind losing the Mint/Account prefixes, since ACCOUNT_TYPE is defined for every Extension.

Agreed on this! And we can certainly keep it as `MintCloseAuthority` for that ambiguity